lib/OpenQA/Utils.pm

```diff
@@ -224,6 +224,27 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+    # After change the needle's properties form to hash, support backward compatibility
+    return $needle unless $needle->{properties};
+
+    my @properties;
+    foreach my $property (@{$needle->{properties}}) {
+        if (ref($property) eq "HASH") {
+            push(@properties, $property);
+            next;
+        }
+        next if ($property ne "workaround");
+
+        my $desc = "";
+        if ($name =~ /\S+\-(bsc\d+|poo\d+)\-\S+/) {
```

Comment:
At least on the openQA-side we could make use of the existing utilities to extract bug references instead of hard-coding strings like 'bsc' and 'poo' here again.

Reply:
Sorry, I do not understand your meaning.  I use this regex code to deal with the already exists needle with workaround property.  If the needle property includes "workaround" string, I will find the related bug number by using regex match needle name and show it in web page.

Reply:
Sorry, I do not understand your meaning. I use this regex code to deal with the already exists needle with workaround property. If the needle property includes "workaround" string, I will find the related bug number by using regex match needle name and show it in description textbox. I am not sure if the operation is needed.

Reply:
I'm saying you should make use of helpers within `lib/OpenQA/Utils.pm`. There is already `find_bugref` for this. If it does not to 100% what you need here you can add a similar helper function but still make use of `keys %bugrefs` instead of hard-coding those abbreviations here again.

Reply:
Thanks for your explanation. I will fix it.